faq_bot/bot/script.py

```diff
@@ -0,0 +1,91 @@
+from dff.script import RESPONSE, TRANSITIONS, Context, Actor, LOCAL
+import dff.script.conditions as cnd
+from dff.script.core.message import Button
+from dff.messengers.telegram import TelegramMessage, TelegramUI, CallbackQuery, ParseMode
+from .model import faq
+
+
+def suggest_similar_questions(ctx: Context, actor: Actor):
+    if ctx.validation:
+        return TelegramMessage()
+    last_request = ctx.last_request
+    if last_request is None:
+        raise RuntimeError("No last requests.")
+    if last_request.annotations is None:
+        raise RuntimeError("No annotations.")
+    similar_questions = last_request.annotations.get("similar_questions")
+    if similar_questions is None:
+        raise RuntimeError("Last request has no text.")
+    if len(similar_questions) == 0:  # question is not similar to any questions
+        return TelegramMessage(
+            text="I don't have an answer to that question. Here's a list of questions I know an answer to:",
+            ui=TelegramUI(
+                buttons=[
+                    Button(text=q, payload=q) for q in faq
+                ]
+            )
+        )
+    else:
+        return TelegramMessage(
+            text="I found similar questions in my database:",
+            ui=TelegramUI(
+                buttons=[
+                    Button(text=q, payload=q) for q in similar_questions
+                ]
+            )
+        )
+
+
+def answer_question(ctx: Context, actor: Actor):
+    if ctx.validation:
+        return TelegramMessage()
+    last_request = ctx.last_request
+    if last_request is None:
+        raise RuntimeError("No last requests.")
+    if last_request.commands is None:
+        raise RuntimeError("No commands")
+    if len(last_request.commands) != 1:
+        raise RuntimeError("Number of Commands != 1")
+    command = last_request.commands[0]
+    if not isinstance(command, CallbackQuery):
+        raise RuntimeError(type(command))
+    return TelegramMessage(text=faq[command.data], parse_mode=ParseMode.HTML)
+
+
+qa_transitions = {
+    ("qa_flow", "suggest_questions"): lambda ctx, actor: ctx.last_request.text is not None,
+    ("qa_flow", "answer_question"): lambda ctx, actor: ctx.last_request.commands is not None,
```

Comment:
can we use regular function with awesome naming?

Reply:
https://github.com/deeppavlov/dialog_flow_demo/pull/1/commits/7b9d6521717f1d9073a1245b8e163d1b67c41618